aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -471,6 +471,31 @@ SparseTensor& mul_out_sparse_cuda(const Tensor& t_, const Tensor& src_, SparseTe
     return _mul_dense_sparse_out(t_, src_, r_);
   }
 
+  // case mul(sparse, sparse) with a 0-dim input.
+  const auto get_vals_from_scalar = [](const SparseTensor& s) -> Tensor {
+    const auto vals = s._values().squeeze(0);
+    if (vals.dim()) {
+      return at::empty({}, vals.options());
+    }
+    return vals;
+  };
+  const auto probably_scalar = (!src_.dim() && src_.is_coalesced())
+    ? src_
+    : (!t_.dim() && t_.is_coalesced())
+    ? t_ : Tensor {};
+  const auto other = is_same_tensor(probably_scalar, t_) ? src_ : t_;
```

Comment:
The nested ternary with compound conditions is a bit much to digest, but I tried and is probably the clearest way to write this. I suggest a comment describing these two assignments so someone does not have to manually pick it apart if they are reading. Same applies to the block in the .cpp